notifications_utils/recipient_validation/phone_number.py

```diff
@@ -219,6 +226,8 @@ def validate_phone_number(self, phone_number: str) -> phonenumbers.PhoneNumber:
         # of those cases separately before we parse with the phonenumbers library
         self._raise_if_phone_number_contains_invalid_characters(phone_number)
 
+        self._raise_if_service_cannot_send_to_landline_but_tries_to(phone_number)
```

Comment:
[Suggestion] I think this check should happen next to where we call `_raise_if_service_cannot_send_to_international_but_tries_to` in init - this is because it has to do with service permissions, not with number being correct or not.